tools/clang/lib/Sema/SemaHLSLDiagnoseTU.cpp

```diff
@@ -295,6 +301,25 @@ clang::FunctionDecl *ValidateNoRecursion(CallGraphWithRecurseGuard &callGraph,
   return nullptr;
 }
 
+class HLSLMethodCallDiagnoseVisitor
+    : public RecursiveASTVisitor<HLSLMethodCallDiagnoseVisitor> {
+public:
+  explicit HLSLMethodCallDiagnoseVisitor(Sema *S, const hlsl::ShaderModel *SM,
+                                         DXIL::ShaderKind EntrySK)
+      : sema(S), SM(SM), EntrySK(EntrySK) {}
+
+  bool VisitCXXMemberCallExpr(CXXMemberCallExpr *CE) {
+    sema->DiagnoseUsedHLSLMethodCall(CE->getMethodDecl(), CE->getExprLoc(), SM,
```

Comment:
The boolean return value for this function is discarded / unused. Are you sure you always want to return true here? Do you not want to use the return value for this change?

Reply:
Return true is OK here.
We already emit the error in DiagnoseUsedHLSLMethodCall.
Maybe we could change DiagnoseUsedHLSLMethodCall to return void?

Reply:
If the only place `DiagnoseUsedHLSLMethodCall` is used we ignore the return value, it seems like it should return void.

Reply:
Updated.